src/components/link/Link.tsx

```diff
@@ -34,46 +34,45 @@ export const Link: FC<ILinkProps> = (props: ILinkProps) => {
     onKeyDown: event => props.onKeyDownLink?.(event, props)
   };
 
-  const lineCenterPos = center(start, end);
-  const linePosition: CSSProperties = {
-    width: len(start, end),
-    borderBottomWidth: props.size,
-    top: lineCenterPos.y,
-    left: lineCenterPos.x,
-    transform: `translate(-50%, -50%) rotate(${deg(start, end)}deg)`
-  };
+  const needClickHelper: boolean =
+    !!props.onClickLink &&
+    (!isFinite(props.size!) || props.size! < CLICK_HELPER_THRESHOLD);
 
+  const lineCenterPos = center(start, end);
+  const lineLength = len(start, end);
   const lineProps: HTMLAttributes<HTMLDivElement> = {
     ...eventHandlers,
+    ...props.rootProps,
+    className: css(LINK_CLASS_LINE, props.rootProps?.className),
     style: {
-      borderBottomColor: props.color,
-      borderBottomStyle: props.lineType,
-      ...linePosition,
-      ...props.lineStyle
-    },
-    tabIndex: props.focusable ? 0 : undefined,
-    "aria-label": props.lineAriaLabel
+      position: "absolute",
+      width: lineLength,
+      height: props.size,
+      top: lineCenterPos.y,
+      left: lineCenterPos.x,
+      transform: `translate(-50%, -50%) rotate(${deg(start, end)}deg)`,
+      display: "flex",
+      alignItems: "center",
+      ...(needClickHelper && {
+        cursor: "pointer",
+        height: CLICK_HELPER_THRESHOLD
+      })
+    }
   };
 
-  const needClickHelper: boolean =
-    !!props.onClickLink &&
-    (!isFinite(props.size!) || props.size! < CLICK_HELPER_THRESHOLD);
-
-  const clickHelperLineStyle: CSSProperties = {
-    ...lineProps.style,
-    opacity: 0,
-    cursor: "pointer",
-    borderBottomWidth: CLICK_HELPER_THRESHOLD
-  };
-  const clickHelperLineProps: React.HTMLAttributes<HTMLDivElement> = {
-    ...eventHandlers,
-    style: clickHelperLineStyle
+  const lineInnerStyles: CSSProperties = {
+    borderBottomColor: props.color,
+    borderBottomStyle: props.lineType,
+    borderBottomWidth: props.size,
+    width: "100%",
+    ...props.lineStyle
   };
 
   return (
     <div className={LINK_CLASS_ROOT} style={props.style}>
```

Comment:
According to the name `rootProps` should be applied to root but it's not.

Reply:
LINK_CLASS_ROOT is only for positioning purpose. It has no actual size. 
![image](https://user-images.githubusercontent.com/15100664/146121792-4fa5b922-c558-44a2-a1ec-39c9ab194b04.png)

Or we name it "lineProps"?